smithy-cli/src/main/java/software/amazon/smithy/cli/commands/InitCommand.java

```diff
@@ -0,0 +1,171 @@
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.cli.commands;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.function.Consumer;
+import software.amazon.smithy.cli.ArgumentReceiver;
+import software.amazon.smithy.cli.Arguments;
+import software.amazon.smithy.cli.Command;
+import software.amazon.smithy.cli.HelpPrinter;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.utils.IoUtils;
+
+public class InitCommand implements Command {
+    private static final String TEMP_DIRECTORY = ".temp";
+    private static final String SMITHY_TEMPLATE_JSON = "smithy-templates.json";
+
+    private final String parentCommandName;
+
+    public InitCommand(String parentCommandName) {
+        this.parentCommandName = parentCommandName;
+    }
+
+    @Override
+    public String getName() {
+        return "init";
+    }
+
+    @Override
+    public String getSummary() {
+        return "Initialize a smithy project by template";
+    }
+
+    @Override
+    public int execute(Arguments arguments, Env env) {
+        arguments.addReceiver(new Options());
+        CommandAction action = HelpActionWrapper.fromCommand(this, parentCommandName, this::run);
+        return action.apply(arguments, env);
+    }
+
+    private int run(Arguments arguments, Env env) {
+        Options options = arguments.getReceiver(Options.class);
+        try {
+            this.executeScript(options.repositoryUrl, options.template, options.directory);
+        } catch (IOException | InterruptedException | URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+
+        return 0;
+    }
+
+    private void executeScript(String repositoryUrl, String templateName, String directory)
+            throws IOException, InterruptedException, URISyntaxException {
+
+        // Use templateName if directory is not specified
+        if (directory == null) {
+            directory = templateName;
+        }
+
+        // Clone the repository without downloading files
+        new ProcessBuilder(
+            "git", "clone", "--filter=blob:none", "--no-checkout",
+                "--depth", "1", "--sparse", repositoryUrl, TEMP_DIRECTORY)
+                .inheritIO().start().waitFor();
+
+        // Download template json file
+        new ProcessBuilder("git", "sparse-checkout", "set", "--no-cone", SMITHY_TEMPLATE_JSON)
```

Comment:
If these ProcessBuilders fail, will the error message be printed to stderr? I noticed only some of them use inheritIO().